libbindgen/src/ir/item.rs

```diff
@@ -668,19 +671,20 @@ impl Item {
     /// type and the parent chain, since it should be consistent.
     pub fn real_canonical_name(&self,
                                ctx: &BindgenContext,
-                               within_namespace: bool,
-                               for_name_checking: bool)
+                               opt: &NameOptions)
                                -> String {
-        let target = ctx.resolve_item(self.name_target(ctx, for_name_checking));
+        debug!("{:?}", opt);
+        let target =
+            ctx.resolve_item(self.name_target(ctx, opt.for_name_checking));
 
         // Short-circuit if the target has an override, and just use that.
-        if !for_name_checking {
+        if !opt.for_name_checking {
             if let Some(other) = target.annotations.use_instead_of() {
                 return other.to_owned();
             }
         }
 
-        let base_name = target.base_name(ctx, for_name_checking);
+        let base_name = target.base_name(ctx, opt.for_name_checking);
```

Comment:
I think it would end up nice if `base_name` and all name-related things took the `NameOptions` rather than a `bool` as well.